### PR TITLE
feat: Add `extract_attachment_info`

### PR DESCRIPTION
**Pull request by mallorih**
This is a draft PR until the previous [PR](https://github.com/Unstructured-IO/unstructured/pull/111/files#diff-a0851d48e2080f3d36ad99c204dbc55a797fa9e2a288cdb2d94051e22e1c9080) is merged.

This PR adds the ability to extract attachment information and payload from an email and save it to disk or store in a python object.

### Testing

```python
 import email
  from unstructured.partition.email import extract_attachment_info

  with open("example-docs/fake-email-attachment.eml", "r") as f:
      msg = email.message_from_file(f)
  attachment_info = extract_attachment_info(msg, output_dir="example-docs")

### CODE REVIEW

File: unstructured/partition/email.py
Comment (MthwRobinson) [inline unstructured/partition/email.py:19]: I'd suggest renaming this to `extract_attachment_info` since we're saving them off to a directory. Once the attachments are in the directory, you can call the appropriate `partition_*` function to do the partitioning.

**Review by MthwRobinson — CHANGES_REQUESTED** [review]
Implementation looks good! Just one naming suggestion. Can we also add a section to the docs showing how to deal with attachments?

File: unstructured/partition/email.py
Comment (mallorih) [reply unstructured/partition/email.py:19]: 😂 I changed it from `extract_attachment_info`, but yes I can change the name.

Comment (MthwRobinson) [reply unstructured/partition/email.py]: 🤣 - it's a great name!

File: docs/source/bricks.rst
Comment (MthwRobinson) [inline docs/source/bricks.rst:85]: Probably self-explanatory from  the name `output_dir`, but I'd also mention that the attachments get saved to `output_dir`.

**Review by MthwRobinson — APPROVED** [review]
Looks good! LGTM after the following changes. Also made one small docs suggestion. Feel free to merge without another review once these items are in:

- Remove `DRAFT` from the PR title
- Add a testing section to the PR summary with test code (can be the same snippet from the docs)
- Run `make tidy` to fix the lint job

**Pull request merged**
